python/ray/util/client/server/dataservicer.py

```diff
@@ -95,6 +137,9 @@ def __init__(self, basic_service: "RayletServicer"):
         )
         # stopped event, useful for signals that the server is shut down
         self.stopped = Event()
+        # Helper for collecting chunks from PutObject calls. Assumes that
+        # that put requests from different objects aren't interleaved.
+        self.chunk_collector = ChunkCollector()
```

Comment:
This could be a problem in a multi-threaded context, right?

Reply:
It should be fine. On the client side if multiple threads try to `put` at once, they'll both need to put their request in the request_queue. Inside `self._requests`, all of the chunks for the first `PutRequest` will be yielded before the next one starts getting chunked.

Reply:
Ahh that makes sense!